app/services/katello/registration_manager.rb

```diff
@@ -151,6 +151,8 @@ def lookup_content_view_environment(activation_keys)
 
       def candlepin_consumer_destroy(host_uuid)
         ::Katello::Resources::Candlepin::Consumer.destroy(host_uuid)
+      rescue RestClient::ResourceNotFound
+        Rails.logger.error(_("Consumer %s was never created") % host_uuid)
```

Comment:
I would change to "attempted to destroy consumer %s from candlepin, but consumer does not exist in candlepin" and make it warn instead of error. The message could raise alarms the way it is now.

Reply:
@beav updated and changed all of these to warn